src/background/telemetry.js

```diff
@@ -2,121 +2,189 @@
-        methods: ["itemDeleted"],
-        objects: ["manage"],
-        extra_keys: ["itemid"],
-      },
-      "itemSelected": {
-        methods: ["itemSelected"],
-        objects: ["manage", "doorhanger"],
-        extra_keys: [],
-      },
-      "addClick": {
-        methods: ["addClick"],
-        objects: ["manage"],
-        extra_keys: [],
-      },
-      "datastore": {
-        methods: ["added", "updated", "deleted"],
-        objects: ["datastore"],
-        extra_keys: ["itemid", "fields"],
-      },
-      "feedback": {
-        methods: ["feedbackClick"],
-        objects: ["manage"],
-        extra_keys: [],
-      },
-      "faq": {
-        methods: ["faqClick"],
-        objects: ["manage"],
-        extra_keys: [],
-      },
-      "itemCopied": {
-        methods: ["usernameCopied", "passwordCopied"],
-        objects: ["manage", "doorhanger"],
-        extra_keys: [],
-      },
-    });
+    browser.telemetry.registerEvents(TELEMETRY_CATEGORY, events);
   } catch (e) {
     if (e.message === "Attempt to register event that is already registered.") {
       // eslint-disable-next-line no-console
-      console.log("telemetry events already registered; skipping registration");
+      console.log("Telemetry events already registered; skipping registration.");
     } else {
       throw e;
     }
   }
+}
 
+function registerScalars() {
+  const scalars = {
+    // Current login count, set when the user opens the doorhanger or manage page.
+    "loginCount": {
+      kind: "count",
+      keyed: false,
+      record_on_release: true,
+    },
+
+    // not implemented yet
+
+    // Total number of logins copied or revealed during the current session
+    // TODO: increment a counter with each copy and each reveal
+    "loginsAccessed": {
+      kind: "count",
+      keyed: false,
+      record_on_release: true,
+    },
+  };
   try {
-    browser.telemetry.registerScalars(TELEMETRY_CATEGORY, {
-      "datastoreCount": {
-        kind: "count",
-        keyed: false,
-        record_on_release: false,
-        expired: false,
-      },
-    });
+    browser.telemetry.registerScalars(TELEMETRY_CATEGORY, scalars);
   } catch (e) {
     if (e.message === "Attempt to register scalar that is already registered.") {
       // eslint-disable-next-line no-console
-      console.log("telemetry scalar already registered; skipping registration");
+      console.log("Telemetry scalars already registered; skipping registration.");
     } else {
       throw e;
     }
   }
 }
 
-const exported = {
-  recordEvent(method, object, extra) {
-    return browser.telemetry.recordEvent(TELEMETRY_CATEGORY, method, object, null, extra || null);
-  },
-  scalarSet(name, value) {
-    return browser.telemetry.scalarSet(`${TELEMETRY_CATEGORY}.${name}`, value);
-  },
-};
+async function recordEvent({ method, object, extra = null, value = null }) {
+  let result;
+  try {
+    // Work around bug 1536877: use an embedded API experiment to submit
```

Comment:
FYI - this may be some time from now. Version uptake is kind of slow and we'd want 95+% of users to be at the threshold version.

Reply:
:-P sad but true